apps/backend/src/lib/search/filter-parser.test.ts

```diff
@@ -0,0 +1,140 @@
+import { describe, test, expect } from "bun:test"
+import { parseQuery } from "./filter-parser"
+
+describe("parseQuery", () => {
+  test("extracts plain search terms", () => {
```

Comment:
Please convert this to a table test us with names to keep the "should ..." pattern going, and please use object matching instead of continuous asserts!